### PR TITLE
fix test environment

### PR DESCRIPTION
**Pull request by lilliCao**
Depends on #1267 (fix ts-test)
Implement

- searching for maven home properly
- remove System.exit
- add check return code from picocli
- use assertJ for test
- do not override managed surefire version

@devonfw/cobigen

### CODE REVIEW

File: cobigen-cli/cli/src/test/java/com/devonfw/cobigen/cli/commandtests/GenerateCommandTest.java [hunk 1]
```diff
@@ -286,12 +286,21 @@ public void generateFromArgsWithQuote() throws IOException {
...
         generatedList.clear();
     }
+
+    /**
+     * This method check the return code from picocli
+     * @param args
+     */
+    private void execute(String[] args) {
+        int exitCode = commandLine.execute(args);
+        assertTrue(exitCode == 0);
```
Comment (maybeec) [inline cobigen-cli/cli/src/test/java/com/devonfw/cobigen/cli/commandtests/GenerateCommandTest.java:304]: please use assertThat(exitCode).isEqualTo(0) to get better error messages

**Review by maybeec — CHANGES_REQUESTED** [review]

File: cobigen-cli/cli/src/test/java/com/devonfw/cobigen/cli/commandtests/GenerateCommandTest.java (hunk 1 above)
Comment (lilliCao) [reply cobigen-cli/cli/src/test/java/com/devonfw/cobigen/cli/commandtests/GenerateCommandTest.java:308]: there is currently no dependency for assertThat in cli

Comment (maybeec) [reply cobigen-cli/cli/src/test/java/com/devonfw/cobigen/cli/commandtests/GenerateCommandTest.java:308]: please then include core-test. Therefore it's implemented.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 2]
```diff
@@ -436,12 +437,12 @@ private void setOutputRootPath(File inputProject) {
                     LOG.info("(" + selectedArtifactNumber + ") " + artifactDescription);
                 } catch (IndexOutOfBoundsException e) {
                     LOG.error("The {} number you have specified is out of bounds!", artifactType);
-                    System.exit(1);
+                    throw (e);
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:440]: We should add a message for the thrown exception here too I think.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 3]
```diff
@@ -436,12 +437,12 @@ private void setOutputRootPath(File inputProject) {
...
                 } catch (NumberFormatException e) {
                     LOG.error(
                         "Error parsing your input. You need to specify {}s using numbers separated by comma (2,5,6).",
                         artifactType);
-                    System.exit(1);
+                    throw (e);
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:445]: We should add a message for the thrown exception here too I think.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 4]
```diff
@@ -541,11 +542,11 @@ private void printFoundArtifacts(ArrayList<GenerableArtifact> possibleArtifacts,
             } catch (NumberFormatException e) {
                 LOG.error("Error parsing your input. You need to specify {}s using numbers separated by comma (2,5,6).",
                     artifactType);
-                System.exit(1);
+                throw (e);
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:545]: We should add a message for the thrown exception here too I think.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 5]
```diff
@@ -541,11 +542,11 @@ private void printFoundArtifacts(ArrayList<GenerableArtifact> possibleArtifacts,
...
 
             } catch (ArrayIndexOutOfBoundsException e) {
                 LOG.error("Error parsing your input. Please give a valid number from the list above.");
-                System.exit(1);
+                throw (e);
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:549]: We should add a message for the thrown exception here too I think.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/UpdateCommand.java [hunk 6]
```diff
@@ -110,7 +114,7 @@ private void userInputPluginSelection(ArrayList<String> userInputPluginForUpdate
      * @param listOfArtifacts
      *            this hold list of artifact id which is need to update
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/UpdateCommand.java:115]: "This holds a list of artifact ids which need to be updated"

**Review by jan-vcapgemini — CHANGES_REQUESTED** [review]

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java (hunk 2 above)
Comment (lilliCao) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:440]: We did. see the log above

Comment (jan-vcapgemini) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:440]: ok.

**Review by jan-vcapgemini — APPROVED** [review]
Looks ok.

**Review by maybeec — APPROVED** [review]

**Comment by jan-vcapgemini** [comment]
Please don't forget to adjust the src/main/resources/pom.xml tsplugin version to the latest one.

**Comment by lilliCao** [comment]
This should be set at next release for all other plugins too

**Pull request merged**